src/prototype_room_utils.js

```diff
@@ -8,6 +8,15 @@ Room.prototype.sortMyRoomsByLinearDistance = function(target) {
   return _.sortBy(Memory.myRooms, sortByLinearDistance);
 };
 
+Room.prototype.nearestRoomName = function(roomsNames, limit) {
```

Comment:
I think I wouldn't pass the `roomsNames` and just use `Memory.myRooms` in the method.

Also you can `_.filter` `myRooms` first, exclude `distance > limit` and the room itself.